### PR TITLE
Feature: Add 1.28 cluster for job generation

### PR DESCRIPTION
**Pull request by mowangdk**
Updates the CSI prow jobs to run on k8s 1.28 and remove 1.24 & 1.25 support.

### CODE REVIEW

File: config/jobs/kubernetes-csi/gen-jobs.sh [hunk 1]
```diff
@@ -24,9 +24,9 @@ base="$(dirname $0)"
 # irrelevant because the prow.sh script will pick a suitable KinD
 # image or build from source.
 k8s_versions="
-1.24
 1.25
 1.26
+1.28
```
Comment (xing-yang) [inline config/jobs/kubernetes-csi/gen-jobs.sh:29]: 1.25->1.26
1.26->1.27

**Comment by mowangdk** [comment]
@xing-yang I guess we can merge this pr now?

File: config/jobs/kubernetes-csi/gen-jobs.sh [hunk 2]
```diff
@@ -37,7 +37,7 @@ deployment_versions="
 "
 
 # The experimental version for which jobs are optional.
-experimental_k8s_version="1.26"
+experimental_k8s_version="1.28"
 
 # The latest stable Kubernetes version for testing alpha jobs
 latest_stable_k8s_version="1.25" # TODO: bump to 1.26 after testing a pull job
```
Comment (xing-yang) [inline config/jobs/kubernetes-csi/gen-jobs.sh:43]: Can you bump this to 1.26 as we've removed 1.25?  Then submit another PR to bump it to 1.28 after testing a pull job after the PR is merged.

**Comment by xing-yang** [comment]
You also need to run the gen_jobs.sh script to update the jobs.  Please update the PR description with what you've run to update the PR. See example here:

https://github.com/kubernetes/test-infra/pull/28255

File: config/jobs/kubernetes-csi/gen-jobs.sh (hunk 2 above)
Comment (mowangdk) [reply config/jobs/kubernetes-csi/gen-jobs.sh:43]: okay~

**Comment by mowangdk** [comment]
@xing-yang done, ptal~

> You also need to run the gen_jobs.sh script to update the jobs. Please update the PR description with what you've run to update the PR. See example here:
> 
> #28255

File: config/jobs/kubernetes-csi/csi-driver-host-path/csi-driver-host-path-config.yaml [hunk 3]
```diff
@@ -2,57 +2,6 @@
...
-        securityContext:
-          privileged: true
-        resources:
-          requests:
-            memory: "9Gi"
-            cpu: 4
-          limits:
-            memory: "9Gi"
-            cpu: 4
   - name: pull-kubernetes-csi-csi-driver-host-path-1-24-on-kubernetes-master
```
Comment (xing-yang) [inline config/jobs/kubernetes-csi/csi-driver-host-path/csi-driver-host-path-config.yaml:5]: I wonder why 1.24 test is still here. Were all the changes in the yaml generated automatically?

Comment (xing-yang) [reply config/jobs/kubernetes-csi/csi-driver-host-path/csi-driver-host-path-config.yaml]: Yes

**Comment by mowangdk** [comment]
@xing-yang all test passed, ptal~

File: config/jobs/kubernetes-csi/gen-jobs.sh [hunk 4]
```diff
@@ -24,23 +24,23 @@ base="$(dirname $0)"
...
+1.28
 "
 
 # The experimental version for which jobs are optional.
-experimental_k8s_version="1.26"
+experimental_k8s_version="1.28"
 
 # The latest stable Kubernetes version for testing alpha jobs
-latest_stable_k8s_version="1.25" # TODO: bump to 1.26 after testing a pull job
+latest_stable_k8s_version="1.26" # TODO: bump to 1.26 after testing a pull job
```
Comment (xing-yang) [inline config/jobs/kubernetes-csi/gen-jobs.sh:43]: We missed updating it for 1.27, so change "bump to 1.26" -> "bump to 1.28".

**Comment by xing-yang** [comment]
Can you add "1.27" to the PR description?  1.27 is also newly added.

**Comment by xing-yang** [comment]
@mowangdk Can you confirm that all the changes in the yamls are generated by the script automatically?

File: config/jobs/kubernetes-csi/gen-jobs.sh (hunk 4 above)
Comment (mowangdk) [reply config/jobs/kubernetes-csi/gen-jobs.sh:43]: @xing-yang I assume the change here was tested and then changed to 1.28 in another pr? According to your previous reply. Did I misunderstand?
image

**Comment by mowangdk** [comment]
> @mowangdk Can you confirm that all the changes in the yamls are generated by the script automatically?

Yes, I only changed config/jobs/kubernetes-csi/gen-jobs.sh. The rest of the changes were made as a result of this script.

**Comment by xing-yang** [comment]
@mowangdk Can you update the PR description with 1.27?  I don't see the change yet.
